pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/name/SimpleSegmentNameGenerator.java

```diff
@@ -66,8 +68,9 @@ public String generateSegmentName(int sequenceId, @Nullable Object minTimeValue,
       SegmentNameUtils.validatePartialOrFullSegmentName(maxTimeValue.toString());
     }
 
-    return JOINER.join(_segmentNamePrefix, minTimeValue, maxTimeValue, _segmentNamePostfix,
-        sequenceId >= 0 ? sequenceId : null, _appendUUIDToSegmentName ? UUID.randomUUID() : null);
+    return JOINER.join(_segmentNamePrefix, _excludeTimestampsInSegmentName ? null : minTimeValue,
```

Comment:
We should put the check before performing the time value check
```suggestion
     if (_excludeTimeInSegmentName) {
       JOINER.join(_segmentNamePrefix, ...);
     } else {
       if (minTimeValue != null) {
       ...
     }
```